lib/ncbo_annotator.rb

```
@@ -781,6 +796,9 @@ def mappings_for_class_ids(class_ids)
       end
 
       def hierarchy_query(class_ids)
+        # mdorf, 12/14/2023: AllegroGraph throws a MalformedQuery exception
+        # if an ID is not of the proper URI format
+        class_ids.select! { |id| id =~ /\A#{URI::regexp}\z/ }
```

RuboCop (1.73)

[warning] `URI::regexp` is obsolete and should not be used. Instead, use `URI::DEFAULT_PARSER.make_regexp`.

(Lint/UriRegexp)